### PR TITLE
[web] stop using deprecated jsonwire web-driver protocol

### PR DESCRIPTION
**Pull request by yjbanov**
Stop using the deprecated jsonwire web-driver protocol. Use the W3C protocol instead.

### CODE REVIEW

**Comment by godofredoc** [comment]
Changes to .ci.yaml file LGTM

File: packages/flutter_tools/lib/src/drive/web_driver_service.dart [hunk 1]
```diff
@@ -260,7 +260,7 @@ Map<String, dynamic> getDesiredCapabilities(
         'chromeOptions': <String, dynamic>{
```
Comment (ditman) [inline packages/flutter_tools/lib/src/drive/web_driver_service.dart:260]:
```suggestion
        'goog:chromeOptions': <String, dynamic>{
```

I seem to need the above for tests to run in my local machine (with flutter drive)

File: packages/flutter_tools/test/general.shard/drive/web_driver_service_test.dart [hunk 2]
```diff
@@ -44,7 +44,7 @@ void main() {
         sync_io.LogType.performance: 'ALL',
       },
       'chromeOptions': <String, dynamic>{
```
Comment (ditman) [inline packages/flutter_tools/test/general.shard/drive/web_driver_service_test.dart:46]:
```suggestion
      'goog:chromeOptions': <String, dynamic>{
```

File: packages/flutter_tools/test/general.shard/drive/web_driver_service_test.dart [hunk 3]
```diff
@@ -72,7 +72,7 @@ void main() {
...
```
Comment (ditman) [inline packages/flutter_tools/test/general.shard/drive/web_driver_service_test.dart:73]:
```suggestion
      'goog:chromeOptions': <String, dynamic>{
```

File: packages/flutter_tools/test/general.shard/drive/web_driver_service_test.dart [hunk 4]
```diff
@@ -101,7 +101,7 @@ void main() {
         sync_io.LogType.performance: 'ALL',
       },
       'chromeOptions': <String, dynamic>{
```
Comment (ditman) [inline packages/flutter_tools/test/general.shard/drive/web_driver_service_test.dart:103]:
```suggestion
      'goog:chromeOptions': <String, dynamic>{
```

File: packages/flutter_tools/lib/src/drive/web_driver_service.dart (hunk 1 above)
Comment (ditman) [inline packages/flutter_tools/lib/src/drive/web_driver_service.dart:260]:
```suggestion
        'goog:chromeOptions': <String, dynamic>{
```

**Review by ditman — COMMENTED** [review]
`'chromeOptions'` needs to be `'goog:chromeOptions'` for tests to run. Otherwise there's an exception when creating the driver (?)

```
InvalidArgumentException (400): invalid argument: unrecognized capability: chromeOptions
#0      parseW3cResponse (package:webdriver/src/handler/w3c/utils.dart:42:9)
#1      W3cSessionHandler.parseCreateResponse (package:webdriver/src/handler/w3c/session.dart:19:21)
#2      InferSessionHandler.parseCreateResponse (package:webdriver/src/handler/infer_handler.dart:106:34)
#3      AsyncRequestClient.send (package:webdriver/src/common/request_client.dart:96:32)
<asynchronous suspension>
#4      createDriver (package:webdriver/async_core.dart:62:19)
<asynchronous suspension>
```

**Review by ditman — APPROVED** [review]
I ran the integration tests in the flutter/packages repository with this change, and everything seems to be passing (at least on Chrome 111):

```terminal
$ ./script/tool_runner.sh drive-examples --web --exclude=google_sign_in,script/configs/exclude_integration_web.yaml

[... A ton of output ...]

------------------------------------------------------------
Run overview:
  packages/camera/camera - ran
  packages/camera/camera_web - ran
  packages/google_identity_services_web - ran
  packages/google_maps_flutter/google_maps_flutter_web - ran
  packages/image_picker/image_picker - ran
  packages/image_picker/image_picker_for_web - ran
  packages/pointer_interceptor - ran
  packages/shared_preferences/shared_preferences - ran
  packages/shared_preferences/shared_preferences_web - ran
  packages/url_launcher/url_launcher - ran
  packages/url_launcher/url_launcher_web - ran
  packages/video_player/video_player - ran
  packages/video_player/video_player_web - ran
  packages/webview_flutter/webview_flutter_web - ran

Ran for 14 package(s)
Skipped 88 package(s)


No issues found!
```

(PS: Skipped google_sign_in because I have changes in flight)

**Comment by yjbanov** [comment]
@ditman Thank you for kicking the tires for this change. Looks like there are good signals that this can stick, so I'm landing it.

**Pull request merged**